lib/tests/test_git.rs

```diff
@@ -2200,7 +2266,10 @@ fn test_reset_head_to_root() {
 
     // Set Git HEAD back to root
     git::reset_head(tx.repo_mut(), &commit1).unwrap();
-    assert!(git_repo.head().is_err());
+    assert!(matches!(
+        git_repo.head().unwrap().kind,
+        gix::head::Kind::Unborn(_)
+    ));
```

Comment:
nit: `.is_unborn()`

btw, I think `.head_id().is_ok()` in this test can be replaced with `!is_unborn()` or `is_detached()`.